### ISSUE TITLE
Editing in WinMerge cause unexpected modification.

### ISSUE THREAD

**Issue by wangyugui** [sdottaka/winmerge-v2#13]
Hi.

Editing in WinMerge cause unexpected modification.
WinMerge version:123(happen in other version too)

```
diff --git a/Trinity b/Trinity
index 3cb9924..ba52e11 100755
--- a/Trinity
+++ b/Trinity
@@ -2617,7 +2624,7 @@ sub show_lit_citation {
         . "Grabherr MG, Haas BJ, Yassour M, Levin JZ, Thompson DA, Amit I, Adiconis X, Fan L,\n"
         . "Raychowdhury R, Zeng Q, Chen Z, Mauceli E, Hacohen N, Gnirke A, Rhind N, di Palma F,\n"
         . "Birren BW, Nusbaum C, Lindblad-Toh K, Friedman N, Regev A.\n"
-        . "Nature Biotechnology 29, 644鈥?52 (2011)\n"
+        . "Nature Biotechnology 29, 644鈥?52 (2011)\n"
         . "Paper: http://www.nature.com/nbt/journal/v29/n7/full/nbt.1883.html\n"
         . "Code:  http://trinityrnaseq.sf.net\n\n\n";
 
hexdump result:
00000000  64 69 66 66 20 2d 2d 67  69 74 20 61 2f 54 72 69  |diff --git a/Tri|
00000010  6e 69 74 79 20 62 2f 54  72 69 6e 69 74 79 0a 69  |nity b/Trinity.i|
00000020  6e 64 65 78 20 33 63 62  39 39 32 34 2e 2e 62 61  |ndex 3cb9924..ba|
00000030  35 32 65 31 31 20 31 30  30 37 35 35 0a 2d 2d 2d  |52e11 100755.---|
00000040  20 61 2f 54 72 69 6e 69  74 79 0a 2b 2b 2b 20 62  | a/Trinity.+++ b|
00000050  2f 54 72 69 6e 69 74 79  0a 40 40 20 2d 32 36 31  |/Trinity.@@ -261|
00000060  37 2c 37 20 2b 32 36 32  34 2c 37 20 40 40 20 73  |7,7 +2624,7 @@ s|
00000070  75 62 20 73 68 6f 77 5f  6c 69 74 5f 63 69 74 61  |ub show_lit_cita|
00000080  74 69 6f 6e 20 7b 0a 20  20 20 20 20 20 20 20 20  |tion {.         |
00000090  2e 20 22 47 72 61 62 68  65 72 72 20 4d 47 2c 20  |. "Grabherr MG, |
000000a0  48 61 61 73 20 42 4a 2c  20 59 61 73 73 6f 75 72  |Haas BJ, Yassour|
000000b0  20 4d 2c 20 4c 65 76 69  6e 20 4a 5a 2c 20 54 68  | M, Levin JZ, Th|
000000c0  6f 6d 70 73 6f 6e 20 44  41 2c 20 41 6d 69 74 20  |ompson DA, Amit |
000000d0  49 2c 20 41 64 69 63 6f  6e 69 73 20 58 2c 20 46  |I, Adiconis X, F|
000000e0  61 6e 20 4c 2c 5c 6e 22  0a 20 20 20 20 20 20 20  |an L,\n".       |
000000f0  20 20 2e 20 22 52 61 79  63 68 6f 77 64 68 75 72  |  . "Raychowdhur|
00000100  79 20 52 2c 20 5a 65 6e  67 20 51 2c 20 43 68 65  |y R, Zeng Q, Che|
00000110  6e 20 5a 2c 20 4d 61 75  63 65 6c 69 20 45 2c 20  |n Z, Mauceli E, |
00000120  48 61 63 6f 68 65 6e 20  4e 2c 20 47 6e 69 72 6b  |Hacohen N, Gnirk|
00000130  65 20 41 2c 20 52 68 69  6e 64 20 4e 2c 20 64 69  |e A, Rhind N, di|
00000140  20 50 61 6c 6d 61 20 46  2c 5c 6e 22 0a 20 20 20  | Palma F,\n".   |
00000150  20 20 20 20 20 20 2e 20  22 42 69 72 72 65 6e 20  |      . "Birren |
00000160  42 57 2c 20 4e 75 73 62  61 75 6d 20 43 2c 20 4c  |BW, Nusbaum C, L|
00000170  69 6e 64 62 6c 61 64 2d  54 6f 68 20 4b 2c 20 46  |indblad-Toh K, F|
00000180  72 69 65 64 6d 61 6e 20  4e 2c 20 52 65 67 65 76  |riedman N, Regev|
00000190  20 41 2e 5c 6e 22 0a 2d  20 20 20 20 20 20 20 20  | A.\n".-        |
000001a0  2e 20 22 4e 61 74 75 72  65 20 42 69 6f 74 65 63  |. "Nature Biotec|
000001b0  68 6e 6f 6c 6f 67 79 20  32 39 2c 20 36 34 34 e2  |hnology 29, 644.|
000001c0  80 93 36 35 32 20 28 32  30 31 31 29 5c 6e 22 0a  |..652 (2011)\n".|
000001d0  2b 20 20 20 20 20 20 20  20 2e 20 22 4e 61 74 75  |+        . "Natu|
000001e0  72 65 20 42 69 6f 74 65  63 68 6e 6f 6c 6f 67 79  |re Biotechnology|
000001f0  20 32 39 2c 20 36 34 34  e2 80 3f 35 32 20 28 32  | 29, 644..?52 (2|
00000200  30 31 31 29 5c 6e 22 0a  20 20 20 20 20 20 20 20  |011)\n".        |
00000210  20 2e 20 22 50 61 70 65  72 3a 20 68 74 74 70 3a  | . "Paper: http:|
00000220  2f 2f 77 77 77 2e 6e 61  74 75 72 65 2e 63 6f 6d  |//www.nature.com|
00000230  2f 6e 62 74 2f 6a 6f 75  72 6e 61 6c 2f 76 32 39  |/nbt/journal/v29|
00000240  2f 6e 37 2f 66 75 6c 6c  2f 6e 62 74 2e 31 38 38  |/n7/full/nbt.188|
00000250  33 2e 68 74 6d 6c 5c 6e  22 0a 20 20 20 20 20 20  |3.html\n".      |
00000260  20 20 20 2e 20 22 43 6f  64 65 3a 20 20 68 74 74  |   . "Code:  htt|
00000270  70 3a 2f 2f 74 72 69 6e  69 74 79 72 6e 61 73 65  |p://trinityrnase|
00000280  71 2e 73 66 2e 6e 65 74  5c 6e 5c 6e 5c 6e 22 3b  |q.sf.net\n\n\n";|
00000290  0a 20 0a                                          |. .|
00000293
```

**Comment by sdottaka** [comment]
Thank you for the report.

I could not reproduce the problem with version 2.14.0+-jp-123.
But I can reproduce the problem with original version 2.14.0.

Do you really use the version 2.14.0+-jp-123?

I think the encoding of the file 'Trinity' is UTF-8. The problem occurs If WinMerge recognizes the file as codepage936(Simplified Chinese).

Here is the screenshots:
[version 2.14.0+-jp-123]
<img width="616" alt="123" src="https://cloud.githubusercontent.com/assets/98126/20858607/0b9c4a1c-b98c-11e6-937c-6ebe771be071.png">

[original version]
<img width="621" alt="orginal" src="https://cloud.githubusercontent.com/assets/98126/20858609/19263116-b98c-11e6-8618-b078b2002100.png">

**Comment by wangyugui** [comment]
Thanks a lot for your info about UTF-8.

I used WinMerge 2.14.0+-jp-123 from TortoiseGit. 
The WinMerge displayed a notice "Information lost dure to encoding error:both file", but I had NOT noticed it.  When I changed it to UTF-8, WinMerge works well.

By the way, The codepage setting of WinMerge is:
default code page:system codepage
detect code page in these files:.html,...     YES
detect code page for text files using mlang.dll   YES
AutoSelect or Auto Select(Chinese)

But WinMerge Failed to detect this text file As UTF-8?

**Comment by sdottaka** [comment]
Could you attach the Trinity file or tell me downloadable URL for reproducing the problem?

**Comment by wangyugui** [comment]
Hi.

I send the file as attachment file.

or you can download it from
https://github.com/trinityrnaseq/trinityrnaseq/blob/master/Trinity


2016-12-04 6:33 GMT+08:00, Takashi Sawanaka <notifications@github.com>:

**Comment by wangyugui** [comment]
By the way, My OS is Windows 7(English User Interface, and 936 as default code page, NOT 932).
![encodingerror](https://cloud.githubusercontent.com/assets/13562993/20863477/242792ba-ba06-11e6-928f-9e68a7e200f8.gif)

**Comment by sdottaka** [comment]
Thank you for the information.
Now I can reproduce it. 

WinMerge detects the encoding of a file using the contents of the first 64Kbytes of the file.
However there is no UTF-8 characters in the contents of the first 64Kbytes of the Trinity file.
As a result, WinMerge could not detects the encoding of the file correctly.

I've fixed the problem and uploaded a new version 2.14.0+-jp-124.
You can download from: 
https://github.com/sdottaka/winmerge-v2/releases/tag/2.14.0%2B-jp-124

**Comment by wangyugui** [comment]
It is fixed in my environment too. 
Thanks a lot for your job.
